helix-view/src/env.rs

```diff
@@ -0,0 +1,25 @@
+use std::fmt::Display;
+
+use crate::Document;
+
+pub fn inject_environment<T>(strs: T, doc: &Document) -> Vec<String>
+where
+    T: Iterator,
+    <T as Iterator>::Item: Display,
+{
+    let path = doc.path().and_then(|p| p.to_str()).unwrap_or_else(|| {
+        log::error!("No $path found for document: {:?}", doc.url());
+        "ENV:NO_PATH"
+    });
```

Comment:
For an example, scratch buffers won't have paths. In cases where `doc.path().is_none()`, I think we should just skip setting the `$pwd`

Reply:
I just accidentally did `:sh echo $file` and it felt weird when it did not work (`$pwd` does work in scratch). I also think logs from failing formatters/lsps will be easier to debug if it actually says something rather than nothing.

All that said, I will defer to you.

I agree with moving the function, on it.

Reply:
I think this would be ok at debug level. We probably want to use `unwrap_or_default` though to have it default to the empty string like what happens when an environment variable is unset

Reply:
`unwrap_or_default` will not let us log the error though? Maybe simply return a default string is better? At least then we got something to go after when some user runs `hx -v` when we ask for logs. Will be a lot easier to spot than having to know if some argument takes a file or not (in the logs from lsp/formatter/whatever).

Reply:
Oh yeah good point, an `unwrap_or_else` that gives logs and gives an empty string would be good I think 👍